pygmt/tests/test_config.py

```diff
@@ -0,0 +1,34 @@
+"""
+Tests for gmt config
+"""
+import pytest
+
+from .. import Figure, config
+
+
+@pytest.mark.mpl_image_compare
+def test_config():
+    """
+    Test if config works globally and locally.
+    """
+    config(FONT_ANNOT_PRIMARY="blue")
+    fig = Figure()
+    fig.basemap(
+        region="0/10/0/10", projection="X10c/10c", frame=["af", '+t"Blue Annotation"']
+    )
+
+    with config(FONT_LABEL="red", FONT_ANNOT_PRIMARY="red"):
+        fig.basemap(
+            region="0/10/0/10",
+            projection="X10c/10c",
+            frame=['xaf+l"red label"', "yaf", '+t"red annotation"'],
+            X="15c",
+        )
+
+    fig.basemap(
+        region="0/10/0/10",
+        projection="X10c/10c",
+        frame=["af", '+t"Blue Annotation"'],
+        X="15c",
+    )
+    return fig
```

Comment:
The changed global config setting doesn't end after the function has finished running (and causes other tests to have blue font...). Is there a way to reset the `FONT_ANNOT_PRIMARY` back to the default setting at the end?

Reply:
I added `config(FONT_ANNOT_PRIMARY="black")` at the end of the test to revert the changes.